commit d0a060be573bfbf8753a15dca35497db5e968bb0
Author: Kristina Martsenko <kristina.martsenko@arm.com>
Date:   Wed Jan 30 12:02:44 2019 +0000

    arm64: add ptrace regsets for ptrauth key management
    
    Add two new ptrace regsets, which can be used to request and change the
    pointer authentication keys of a thread. NT_ARM_PACA_KEYS gives access
    to the instruction/data address keys, and NT_ARM_PACG_KEYS to the
    generic authentication key. The keys are also part of the core dump file
    of the process.
    
    The regsets are only exposed if the kernel is compiled with
    CONFIG_CHECKPOINT_RESTORE=y, as the only intended use case is
    checkpointing and restoring processes that are using pointer
    authentication. (This can be changed later if there are other use
    cases.)
    
    Reviewed-by: Dave Martin <Dave.Martin@arm.com>
    Signed-off-by: Kristina Martsenko <kristina.martsenko@arm.com>
    Signed-off-by: Catalin Marinas <catalin.marinas@arm.com>

diff --git a/Documentation/arm64/pointer-authentication.txt b/Documentation/arm64/pointer-authentication.txt
index a25cd21290e9..5baca42ba146 100644
--- a/Documentation/arm64/pointer-authentication.txt
+++ b/Documentation/arm64/pointer-authentication.txt
@@ -78,6 +78,11 @@ bits can vary between the two. Note that the masks apply to TTBR0
 addresses, and are not valid to apply to TTBR1 addresses (e.g. kernel
 pointers).
 
+Additionally, when CONFIG_CHECKPOINT_RESTORE is also set, the kernel
+will expose the NT_ARM_PACA_KEYS and NT_ARM_PACG_KEYS regsets (struct
+user_pac_address_keys and struct user_pac_generic_keys). These can be
+used to get and set the keys for a thread.
+
 
 Virtualization
 --------------